### PR TITLE
Report CI results back to GitHub

### PR DESCRIPTION
**Pull request by pitr-ch**

### CODE REVIEW

**Comment by pitr-ch** [comment]
The builds are added incrementally in pending state when they start executing. To fix it we would have to add it to the build enumerator.

**Comment by pitr-ch** [comment]
I've included a tool for easy push to bitbucket of current PR. It looks up the PR's head based on current git HEAD and pushes it to bitbucket starting the CI run. Alternatively you can tell it the PR number.

File: tool/jt.rb [hunk 1]
```diff
@@ -750,6 +752,42 @@ def cextc(cext_dir, test_gem=false, *clang_opts)
...
+    bb       = remote_urls.find { |r, u| u.include? 'ol-bitbucket' }.first
+
+    pr_number = args.first
+    if pr_number
+      github_pr_branch = "#{upstream}/pr/#{pr_number}"
+    else
+      fetch     = "+refs/pull/*/head:refs/remotes/#{upstream}/pr/*"
+      out, _err = raw_sh("git config --get-all remote.#{upstream}.fetch", capture: true)
+      raw_sh "git config --add remote.#{upstream}.fetch \"#{fetch}\"" unless out.include? fetch
+      raw_sh "git fetch #{upstream}"
```
Comment (eregon) [inline tool/jt.rb]: What is this complex logic?
Shouldn't it be as simple as
```
git push ol my-branch
# Or
git push ol my-branch:github/pr/mybranch
```
I think the namespacing is not necessary and make branches longer than needed if this our primary way to develop.

Comment (pitr-ch) [inline tool/jt.rb]: The intention is that we just maintain our branches on GitHub and use the tool to make a copy on bb to trigger the CI. I expect that in gated PR driven development model, we won't push named branches to bb manually any more, we'll just need a copy of the PR on bb. `guthub/pr/<number>` is very clear where to find the source, avoids name collisions, and does not require maintenance (can be just left there).

File: ci.hocon [hunk 2]
```diff
@@ -13,13 +15,48 @@ common-base: {
...
+            "description": "The ${BUILD_NUMBER} is being executed.",
+            "context": "CI/${BUILD_NAME}" }
+          """],
     [mx, build, --force-javac]
   ]
 
+  on_success: [
+    ${common-base.curl-status-update} ["""
+          { "state": "success",
+            "description": "The ${BUILD_NUMBER} succeded.",
```
Comment (eregon) [inline ci.hocon]: `The build ${BUILD_NUMBER} succeded.`
or just
`${BUILD_NUMBER} succeded.`

Comment (eregon) [inline ci.hocon]: `passed` is also a bit shorter and matches Travis' status.

**Comment by pitr-ch** [comment]
FYI: 
* If a CI build fails in setup it will remain in pending state here, because on_failure is not executed for failures occurred in setup.
* Check the number of tests, they appear as they start executing so there might be test results missing for tests which did not yet started.

**Pull request merged**